sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsDataWriteFactory.scala

```diff
@@ -12,53 +12,49 @@ import org.apache.spark.sql.connector.write.{DataWriter, DataWriterFactory, Writ
 import org.apache.spark.sql.types.StructType
 
 // scalastyle:off multiple.string.literals
-class CosmosDataWriteFactory(userConfig: Map[String, String],
-                             inputSchema: StructType,
-                             cosmosClientStateHandle: Broadcast[CosmosClientMetadataCachesSnapshot])
+private class ItemsDataWriteFactory(userConfig: Map[String, String],
+                            inputSchema: StructType,
+                            cosmosClientStateHandle: Broadcast[CosmosClientMetadataCachesSnapshot])
   extends DataWriterFactory
     with CosmosLoggingTrait {
   logInfo(s"Instantiated ${this.getClass.getSimpleName}")
 
   override def createWriter(i: Int, l: Long): DataWriter[InternalRow] = new CosmosWriter(inputSchema)
 
-  class CosmosWriter(inputSchema: StructType) extends DataWriter[InternalRow] {
+  private class CosmosWriter(inputSchema: StructType) extends DataWriter[InternalRow] {
     logInfo(s"Instantiated ${this.getClass.getSimpleName}")
 
-    val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(userConfig)
-    val cosmosTargetContainerConfig = CosmosContainerConfig.parseCosmosContainerConfig(userConfig)
-    val cosmosWriteConfig = CosmosWriteConfig.parseWriteConfig(userConfig)
+    private val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(userConfig)
+    private val cosmosTargetContainerConfig = CosmosContainerConfig.parseCosmosContainerConfig(userConfig)
+    private val cosmosWriteConfig = CosmosWriteConfig.parseWriteConfig(userConfig)
 
     // TODO moderakh: this needs to be shared to avoid creating multiple clients
-    val builder = new CosmosClientBuilder()
+    private val builder = new CosmosClientBuilder()
       .key(cosmosAccountConfig.key)
       .endpoint(cosmosAccountConfig.endpoint)
       .consistencyLevel(ConsistencyLevel.EVENTUAL)
 
     SparkBridgeInternal.setMetadataCacheSnapshot(builder, cosmosClientStateHandle.value)
-    val client = builder.buildAsyncClient();
+    private val client = builder.buildAsyncClient()
 
-    val container = client.getDatabase(cosmosTargetContainerConfig.database)
+    private val container = client.getDatabase(cosmosTargetContainerConfig.database)
       .getContainer(cosmosTargetContainerConfig.container)
 
-    val containerDefinition = container.read().block().getProperties
-    val partitionKeyDefinition = containerDefinition.getPartitionKeyDefinition
+    private val containerDefinition = container.read().block().getProperties
+    private val partitionKeyDefinition = containerDefinition.getPartitionKeyDefinition
 
     override def write(internalRow: InternalRow): Unit = {
       val objectNode = CosmosRowConverter.fromInternalRowToObjectNode(internalRow, inputSchema)
 
-      // TODO modreakh investigate if we should also support point write in non-blocking way
+      // TODO moderakh investigate if we should also support point write in non-blocking way
```

Comment:
haha this one was interesting :-)

Reply:
Intelli-J dictionary found it :-) The right one I had added to the dictionary :-)